lib/transcoder.ex

```diff
@@ -1,9 +1,45 @@
 defmodule Transcoder do
-  @moduledoc """
-  Transcoder keeps the contexts that define your domain
-  and business logic.
+  use GenServer
+  use TranscoderWeb, :verified_routes
+  alias Transcoder.Model.Video
 
-  Contexts are also responsible for managing your data, regardless
-  if it comes from the database, an external API or others.
-  """
+  def start_link do
+    GenServer.start_link(__MODULE__, nil)
+  end
+
+  def transcode(pid, video, resolution) do
+    GenServer.cast(pid, {:transcode, video, resolution, self()})
+  end
+
+  @impl true
+  def init(_) do
+    {:ok, nil}
+  end
+
+  @impl true
+  def handle_cast({:transcode, video, resolution, from}, _state) do
+    scaled_video = do_transcode(video, resolution)
+    send(from, {:transcode_result, scaled_video})
+    {:noreply, nil}
+  end
+
+  defp do_transcode(original, resolution) do
```

Comment:
A guard stating the supported resolutions might be helpful. In future elixir releases we'll gain safety in [set-theoretic types](https://elixir-lang.org/blog/2024/06/12/elixir-v1-17-0-released/).

Reply:
Added in `transcode` since it's public function.